### ISSUE TITLE
IR sensor on AIN1 not going to zero

### ISSUE THREAD

**Issue by martykube** [pgmmpk/beaglebone_pru_adc#7]
Hi Mike,

I'm trying to calibrate my IR sensors and I have a problem were the left rear sensor (on AIN1) does not go to zero.  Here is what I see when running qb_server.py with nothing close to the sensors:

```
Sent    : $IRVAL?*
Received: [757.033691406, 17.7177734375, 2.66845703125, 0.99951171875, 3.09228515625]
```

To troubleshoot this I tried a few things.  I first swapped physical sensors on AIN1 all have the same issue (but on the other AIN).  I've also verified that the voltage from the IR sensors at the voltage divider does go to 0V when nothing is in proximity.

I ran the examples/basic.py to check what the PRU is reading and here is a few lines of the output:

```
16 (3319, 720, 199, 0, 0, 0, 0, 3504)
47 (3336, 741, 230, 18, 59, 0, 0, 3508)
54 (3331, 715, 192, 0, 0, 0, 0, 3506)
61 (3322, 721, 192, 0, 0, 0, 0, 3504)
67 (3298, 687, 167, 0, 24, 0, 0, 3500)
74 (3310, 685, 171, 0, 6, 0, 0, 3499)
80 (3305, 689, 172, 0, 0, 0, 0, 3504)
87 (3299, 691, 169, 0, 0, 0, 0, 3499)
93 (3311, 699, 169, 0, 0, 0, 0, 3497)
99 (3312, 700, 172, 0, 19, 0, 0, 3504)
106 (3302, 685, 171, 0, 0, 0, 0, 3499)
112 (3312, 693, 172, 0, 0, 0, 0, 3497)
118 (3311, 704, 167, 0, 0, 0, 0, 3500)
125 (3317, 727, 197, 0, 46, 0, 0, 3504)
131 (3331, 724, 200, 0, 52, 0, 0, 3504)
138 (3332, 642, 147, 0, 0, 0, 0, 3502)
144 (3302, 685, 171, 0, 0, 0, 0, 3503)
```

I did not have the encoders wired up, so I'm guessing that is the first and last value, 3319 and 3504 on the first line.  I'm guessing the second column is the AIN1 reading.

I also have to say I saw at least once where this just went away and the sensor did correctly zero, and the after a reboot and the issue returned. 

I did have to swap AIN1 and AIN3 on the left sensor in config.py to match my wiring.

The sensor does respond to changes in proximity and appears to go up to the max voltage when the proximity is high.

Let me know if you have suggestions on how to troubleshoot.

Thanks,
Marty

**Comment by pgmmpk** [comment]
Hi Marty,

Replying from memory, double check the values please.

I had the same issue in my BB. I tracked it down to the fact that 1K
resistor could not bring ADC input pin to zero. My experiment was rather
simple: without even bothering about connecting IR sensors --- just use the
resistor to connect pin to the ground and watch it NOT generating zero (you
can use any software - mine of AdaFruit to measure it).

I ended up changing the voltage divider by using 100 Ohm + 1 kOhm
resistors, instead of 1 kOhm + 10 kOhm. I also believe it was substantiated
by some ADC spec research, but my question regarding this issue was never
answered on the course forum.

In short, I used voltage divider with resistor values 10 times lower than
in specs. I am not very sure if specs were saying to use 1k+10k (and I made
them 0.1k+1k) or specs were 10k+100k (and I have to make them 1k+10k).
Please check by experimenting...

Be careful! IR voltage range is 5V, and ADC input pins will burn if they
get anything larger than 1.8V! Double-check divider before connecting to
the input pin. Make sure that if you connect divider input to +5V divider
output is about +0.5V.

Let me know if you need more details - I'll dig them up.

Best,
-Mike
